opentelemetry-sdk/tests/metrics/integration_test/test_console_exporter.py

```diff
@@ -26,7 +27,7 @@ class TestConsoleExporter(TestCase):
     def test_console_exporter(self):
 
         try:
-            exporter = ConsoleMetricExporter()
+            exporter = ConsoleMetricExporter(out=open(devnull, "w"))
```

Comment:
I don't think this test ever actually calls the exporter because we're not calling `provider.shutdown()`. So it's not actually testing `ConsoleMetricExporter` at all

Maybe this test would be better to capture the IO with a [StringIO](https://docs.python.org/3/library/io.html#io.StringIO) and assert that `ConsoleMetricExporter` wrote something. I think that would show the existing bug here too

Reply:
We do call the exporter when we run this test:

```
ocelotl@hilleman:~/github/ocelotl/opentelemetry-python$ tox -e py39-opentelemetry-sdk -- opentelemetry-sdk/tests/metrics/integration_test/test_console_exporter.py -s
py39-opentelemetry-sdk installed: asgiref==3.5.2,attrs==22.1.0,Deprecated==1.2.13,flaky==3.7.0,iniconfig==1.1.1,opentelemetry-api @ file:///home/ocelotl/github/ocelotl/opentelemetry-python/opentelemetry-api,opentelemetry-sdk @ocelotl 

...

    Uninstalling opentelemetry-test-utils-0.33b0:
      Successfully uninstalled opentelemetry-test-utils-0.33b0
Successfully installed opentelemetry-api-1.12.0 opentelemetry-sdk-1.12.0 opentelemetry-semantic-conventions-0.33b0 opentelemetry-test-utils-0.33b0
py39-opentelemetry-sdk run-test: commands[0] | pytest metrics/integration_test/test_console_exporter.py -s
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-7.1.2, pluggy-1.0.0 -- /home/ocelotl/github/ocelotl/opentelemetry-python/.tox/py39-opentelemetry-sdk/bin/python
cachedir: .tox/py39-opentelemetry-sdk/.pytest_cache
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/ocelotl/github/ocelotl/opentelemetry-python, configfile: pyproject.toml
plugins: benchmark-3.4.1, flaky-3.7.0
collecting ... collected 1 item

metrics/integration_test/test_console_exporter.py::TestConsoleExporter::test_console_exporter PASSED

============================== 1 passed in 0.08s ===============================
{"resource_metrics": [{"resource": {"attributes": {"telemetry.sdk.language": "python", "telemetry.sdk.name": "opentelemetry", "telemetry.sdk.version": "1.12.0", "service.name": "unknown_service"}, "schema_url": ""}, "scope_metrics": [{"scope": {"name": "test_console_exporter", "version": "", "schema_url": ""}, "metrics": [{"name": "test", "description": "", "unit": "", "data": {"data_points": [{"attributes": {}, "start_time_unix_nano": 1660750784653839641, "time_unix_nano": 1660750784654821322, "value": 1}], "aggregation_temporality": 2, "is_monotonic": true}}], "schema_url": ""}], "schema_url": ""}]}
___________________________________ summary ____________________________________
  py39-opentelemetry-sdk: commands succeeded
  congratulations :)
```

Notice the `-s` option was used and the output includes this which comes from the console exporter:

```
{"resource_metrics": [{"resource": {"attributes": {"telemetry.sdk.language": "python", "telemetry.sdk.name": "opentelemetry", "telemetry.sdk.version": "1.12.0", "service.name": "unknown_service"}, "schema_url": ""}, "scope_metrics": [{"scope": {"name": "test_console_exporter", "version": "", "schema_url": ""}, "metrics": [{"name": "test", "description": "", "unit": "", "data": {"data_points": [{"attributes": {}, "start_time_unix_nano": 1660750784653839641, "time_unix_nano": 1660750784654821322, "value": 1}], "aggregation_temporality": 2, "is_monotonic": true}}], "schema_url": ""}], "schema_url": ""}]}
```

I agree that this particular test is quite simple, since it only makes sure that a simple use case works and no unexpected exceptions are raised. We could make it better by comparing the output, but that is not the intention of this PR, I just want to remove the error message from our testing output.

Reply:
> I agree that this particular test is quite simple, since it only makes sure that a simple use case works

Check out this branch, the test is still passing https://github.com/open-telemetry/opentelemetry-python/compare/main...aabmass:opentelemetry-python:consoleexportertest-notworking?expand=1

```console
$ tox -e py310-opentelemetry-sdk -- opentelemetry-sdk/tests/metrics/integration_test/test_console_exporter.py -s
...
py310-opentelemetry-sdk run-test: commands[0] | pytest metrics/integration_test/test_console_exporter.py -s
============================================================================================================================================================ test session starts =============================================================================================================================================================
platform linux -- Python 3.10.5, pytest-7.1.2, pluggy-1.0.0 -- /usr/local/google/home/aaronabbott/repo/opentelemetry-python/.tox/py310-opentelemetry-sdk/bin/python
cachedir: .tox/py310-opentelemetry-sdk/.pytest_cache
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /usr/local/google/home/aaronabbott/repo/opentelemetry-python, configfile: pyproject.toml
plugins: benchmark-3.4.1, flaky-3.7.0
collected 1 item

metrics/integration_test/test_console_exporter.py::TestConsoleExporter::test_console_exporter PASSED

============================================================================================================================================================= 1 passed in 0.13s ==============================================================================================================================================================
Exception while exporting metrics This should be caught in test
Traceback (most recent call last):
  File "/usr/local/google/home/aaronabbott/repo/opentelemetry-python/.tox/py310-opentelemetry-sdk/lib/python3.10/site-packages/opentelemetry/sdk/metrics/_internal/export/__init__.py", line 493, in _receive_metrics
    self._exporter.export(
  File "/usr/local/google/home/aaronabbott/repo/opentelemetry-python/.tox/py310-opentelemetry-sdk/lib/python3.10/site-packages/opentelemetry/sdk/metrics/_internal/export/__init__.py", line 150, in export
    raise Exception("This should be caught in test")
Exception: This should be caught in test
__________________________________________________________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________________________________________________________
  py310-opentelemetry-sdk: commands succeeded
  congratulations :)
```

Reply:
Ok, updating the name of the PR to reflect its increased scope and updated the test too.